### ISSUE TITLE
SAME_UPPER autopad method not implemented. 

### ISSUE THREAD

**Issue by SurenderHarsha** [ENOT-AutoDL/onnx2torch#144]
Hi, I have a custom UNET in ONNX and I see that when I try to convert the model into pytorch, the method mentioned above is not implemented. Would it be difficult to implement? or is there an alternative solution.

**Comment by JohnMasoner** [comment]
Hey @SurenderHarsha 

Could you provide ONNX to reproduce your issue?

Maybe you could try the MONAI repo UNET.

**Comment by SurenderHarsha** [comment]
Hi, I cannot provide the UNET, but I can tell you that each convolutional layer has an auto_pad op at the end of the node with SAME_UPPER, also the network was a tensorflow 1 network which I converted to ONNX using tf2onnx package and now I am looking to convert it into pytorch while making sure the outputs between the networks do not differ. The network I use is a custom production model.

**Comment by SurenderHarsha** [comment]
![image](https://user-images.githubusercontent.com/13507016/221403292-9add63cc-acec-4a79-a54c-9197b1d084cc.png)
I do see that in the padding.py code that auto_pad , SAME_UPPER raises a NotImplementedError.

**Comment by JohnMasoner** [comment]
Hey @SurenderHarsha ,

The reason for this problem is because SAME_UPPER is not implemented.

I need to take the time to reproduce this.(Maybe

**Comment by SurenderHarsha** [comment]
We made our own custom implementation for the auto_pad same upper, this is not generalizable. So closing this issue for now.

**Comment by LamMoh1** [comment]
@SurenderHarsha could you please share the code? I'll try reimplement for my model.

**Comment by Ralfons-06** [comment]
Ran into the same problem. Since i assume `SAME_UPPER` is pads input equally as `same`, i could fix the issue by changing following code snippet from the `utils.padding.onnx_auto_pad_to_torch_padding()` function from:

```
if auto_pad == 'VALID':
    return 0, None

if auto_pad in ('SAME_UPPER', 'SAME_LOWER'):
    raise NotImplementedError(f'"{auto_pad}" auto_pad is not implemented')
``` 

to:

```
if auto_pad == 'VALID':
    return 0, None

if auto_pad == 'SAME_UPPER':
    return "same", None

if auto_pad in ('SAME_LOWER'):
    raise NotImplementedError(f'"{auto_pad}" auto_pad is not implemented')
``` 

I tested the converted model and it runs perfectly fine.
